contracts/src/Assets.sol

```diff
@@ -96,6 +104,11 @@ library Assets {
             revert InvalidToken();
         }
 
+        if ($.registeredNativeTokens[token]) {
```

Comment:
Just a note here, if token creation fails on the substrate side mapping will be populated and stop a retry by re-submitting the registerToken. Someone would need to fix it manually on the substrate end.

Reply:
good point! will remove this check.